src/chainstate/stacks/miner.rs

```diff
@@ -77,12 +99,17 @@ impl<'a> StacksMicroblockBuilder<'a> {
         anchor_block_consensus_hash: ConsensusHash,
         chainstate: &'a mut StacksChainState,
         burn_dbconn: &'a dyn BurnStateDB,
-        initial_cost: ExecutionCost,
-        bytes_so_far: u64,
     ) -> Result<StacksMicroblockBuilder<'a>, Error> {
+        let runtime = if let Some(ref unconfirmed_state) = chainstate.unconfirmed_state.as_ref() {
+            MicroblockMinerRuntime::from_unconfirmed_state(unconfirmed_state)
```

Comment:
Style nit: one of the reasons for the `From` trait is to avoid type stuttering, e.g.,

```rust
MicroblockMinerRuntime::from(unconfirmed_state)
```
vs.
```rust
MicroblockMinerRuntime::from_unconfirmed_state(unconfirmed_state)
```

Reply:
This isn't tractable -- `MicroblockMinerRuntime::from()` would take a `UnconfirmedState` instance, instead of a reference.  In doing so, we'd need to implement `Clone` for a `ClarityInstance`, as well as a `MarfedKV`, a `MARF`, a `SqliteConnection`, etc.  So, I added `from_unconfirmed_state()` which takes a `&UnconfirmedState` in order to avoid this.  Is that alright with you?

Reply:
While leaving it as is would be fine, you can define a trait `From<&UnconfirmedState>` which would have the same borrow semantics as this function.

Reply:
:man_facepalming: ah, right, forgot you could do that.  Will use that instead.

Reply:
Addressed